content/en/docs/concepts/services-networking/dual-stack.md

```diff
@@ -68,47 +72,169 @@ An example of an IPv6 CIDR: `fdXY:IJKL:MNOP:15::/64` (this shows the format but
+  * The cluster must have dual-stack networking configured.
+
+If you would like to define which IP family to use for single stack or define the order of IP families for dual-stack, you can choose the address families by setting an optional field, `.spec.ipFamilies`, on the Service. 
 
 {{< note >}}
-The default address family for your cluster is the address family of the first service cluster IP range configured via the `--service-cluster-ip-range` flag to the kube-controller-manager.
+The `.spec.ipFamilies` field is immutable because the `.spec.ClusterIP` cannot be reallocated on a Service that already exists. If you want to change `.spec.ipFamilies`, delete and recreate the Service.
 {{< /note >}}
 
-You can set `.spec.ipFamily` to either:
+You can set `.spec.ipFamilies` to any of the following array values:
+
+- `["IPv4"]`
+- `["IPv6"]`
+- `["IPv4","IPv6"]` (dual stack)
+- `["IPv6","IPv4"]` (dual stack)
+
+The first family you list is used for the legacy `.spec.ClusterIP` field.
+
+### Dual-stack Service configuration scenarios
+
+These examples demonstrate the behavior of various dual-stack Service configuration scenarios.
 
-   * `IPv4`: The API server will assign an IP from a `service-cluster-ip-range` that is `ipv4`
-   * `IPv6`: The API server will assign an IP from a `service-cluster-ip-range` that is `ipv6`
+#### Dual-stack options on new Services
 
-The following Service specification does not include the `ipFamily` field. Kubernetes will assign an IP address (also known as a "cluster IP") from the first configured `service-cluster-ip-range` to this Service.
+1. This Service specification does not explicitly define `.spec.ipFamilyPolicy`. When you create this Service, Kubernetes assigns a cluster IP for the Service from the first configured `service-cluster-ip-range` and sets the `.spec.ipFamilyPolicy` to `SingleStack`. ([Services without selectors](/docs/concepts/services-networking/service/#services-without-selectors) and [headless Services](/docs/concepts/services-networking/service/#headless-services) with selectors will behave in this same way.)
 
 {{< codenew file="service/networking/dual-stack-default-svc.yaml" >}}
 
-The following Service specification includes the `ipFamily` field. Kubernetes will assign an IPv6 address (also known as a "cluster IP") from the configured `service-cluster-ip-range` to this Service.
+1. This Service specification explicitly defines `PreferDualStack` in `.spec.ipFamilyPolicy`. When you create this Service on a dual-stack cluster, Kubernetes assigns both IPv4 and IPv6 addresses for the service. The control plane updates the `.spec` for the Service to record the IP address assignments. The field `.spec.ClusterIPs` is the primary field, and contains both assigned IP addresses; `.spec.ClusterIP` is a secondary field with its value calculated from `.spec.ClusterIPs`.
```

Comment:
Hi @bridgetkromhout .
If you have time, I think the list numbering will work if you indent the "codenew" shortcode (4 spaces).

Reply:
Thanks. Let's catch the refactoring after we get all the content locked!

Reply:
@kbhawkey @bridgetkromhout - i saw some lints mentioned in the project structure.will one such exists help on these here?